### PR TITLE
Document Android system tracing

### PR DESCRIPTION
**Pull request by keirthana**
A first draft of the system tracing topic leveraged from the internal docs.

This will need additional changes to `index.md` and `howto/android/landing.md` that are part of this PR: https://github.com/canonical/anbox-cloud-docs/pull/33/files

### CODE REVIEW

**Comment by keirthana** [comment]
@adglkh I pulled out the content for system tracing with some modifications into this PR. Please leave your feedback.

File: howto/android/system-tracing.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+This guide describes how to enable system tracing in Android using Anbox Cloud. 
+
+## Enable Android System Tracing
+
+1. To enable Android system tracing, add the `android.enable_tracing` flag into the feature file. In the Android container, run:
```
Comment (keirthana) [inline howto/android/system-tracing.md:5]: > In the Android container

I changed the "development container" to "Android container". Is it correct? Or should this be "In the Anbox instance"?

File: howto/android/system-tracing.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ 
+    For instance, to enable the graphics category, add the following to the `/var/lib/anbox/android.json` file:
+
+    ```json
+      {
+        "hardware": "ranchu",
+        "api_level": 31,
+        "properties": [ "debug.atrace.tags.enableflags=2" ]
+      }
+    ```
```
Comment (ajanon) [inline howto/android/system-tracing.md:20]: I think we need to reboot the instance after changing the android.json file, but we need to double-check this with @morphis or @adglkh.

File: howto/android/system-tracing.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ 3. Launch an Anbox session and initiate tracing via the HTTP endpoint. This will not only trace the Anbox session but also capture events aggregated under the category specified in the above system property.: 
+
+    ```bash
+    curl --unix-socket /run/user/1000/anbox/sockets/api.unix --request POST --data '{"enable":true}' s/1.0/tracing
+      ```   
+     After your desired tracing time, stop the tracing:
+    ```bash
+    curl --unix-socket /run/user/1000/anbox/sockets/api.unix --request POST --data '{"enable":false}' s/1.0/tracing
+      ```   
+      The resulting trace file is available in the response once tracing is disabled. The trace file should look similar to:
```
Comment (ajanon) [inline howto/android/system-tracing.md:30]: So...  The file is not available then, the _path_to the file is available in the response.

File: howto/android/system-tracing.md (hunk 1 above)
Comment (ajanon) [reply howto/android/system-tracing.md:5]: Android runs in a container nested in an instance (container or VM), so that seems accurate to me.

**Review by ajanon — CHANGES_REQUESTED** [review]

**Comment by keirthana** [comment]
Putting this on hold as there is some dilemma about whether to include this for 1.21.

**Comment by morphis** [comment]
> Putting this on hold as there is some dilemma about whether to include this for 1.21.

What I agreed with @adglkh was that we don't document this yet officially as it's use is rather harry and we haven't completed the implementation yet to be fully integrated to what people expect from Android.

**Comment by keirthana** [comment]
@morphis I have updated the RN with the feedback so far. Could you have another look?

I think we need to add [AC-2071](https://warthogs.atlassian.net/browse/AC-2071), could you help me with the user impact for this one? We discussed this in SU before my vacation and I couldn't follow up with you on this.

**Comment by keirthana** [comment]
Closing the PR without merging as this is not ready yet. (As discussed and confirmed with the team as on Feb 13, 2024)

**Pull request closed without merging**